### ISSUE TITLE
Doc: DeferredSemaphore correction

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8024]
|<img alt="fr-anthony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| fr-anthony reported|
|-|-|
|Trac ID|trac#8024|
|Type|enhancement|
|Created|2015-09-23 18:41:33Z|
|Branch|https://github.com/twisted/twisted/tree/deferredsemaphore-docs-8024|

Now the documentation for `twisted.defer.DeferredSemaphore` refers to the correct instance attributes.

Attachments:

* [8024-deferredsemaphore-docs.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/860/8601a89720c0b2137d187f041d6ba096f16906f8/1dc028a31a732e27bd2f4cb0a9499dacc2837fbc.patch) (1348 bytes) - added by fr-anthony on 2015-09-23 18:43:01Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8024 8024
type__enhancement enhancement
reporter__fr_anthony fr-anthony
priority__low low
milestone__ 
branch__branches_deferredsemaphore_docs_8024 branches/deferredsemaphore-docs-8024
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__doc doc
time__1443033693758382 1443033693758382
changetime__1443075956534182 1443075956534182
version__None None
owner__fr_anthony fr-anthony

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [45701]) Branching to deferredsemaphore-docs-8024.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to fr-anthony|
|-|-|

Changes look good. Thanks!

Just a minor comment regarding the release notes fragment.

I think that twisted.defer.DeferredSemaphore should be twisted.internet.defer.DeferredSemaphore


Maybe instead of the `refers to the correct instance attributes.` we should have

Now the documentation for `twisted.internet.defer.DeferredSemaphore` describes to actual usage for `limit` and `tokens` instance attributes.

What do you think?

-----------

As soon as the test will pass I will look into updating the news file and  merging this.

Many thanks for your contribution!

**Comment by twisted-trac** [comment]
|<img alt="fr-anthony's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|fr-anthony commented|
|-|-|

You're absolutely right about `twisted.internet.defer` and I'm okay with the release-note change provided it's corrected to "describes _the_ actual usage". :)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

True.
Will update and merge.

It is past midnight here so I will do the merge tomorrow, just to make sure I am not doing anything stupid.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set status to `closed`|
|-|-|

(In [45710]) Merge deferredsemaphore-docs-8024: Fix t.i.defer.DeferredSemaphore docstring.

Author: fr-anthony
Reviewer: adiroiban
Fixes: [#8024](https://github.com/twisted/twisted/issues/8024)
